FoodTracker/MealViewController.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  ViewController.swift
+//  FoodTracker
+//
+//  Created by Josip Glasovac on 05/10/2018.
+//  Copyright © 2018 Josip Glasovac. All rights reserved.
+//
+
+import UIKit
+import os.log
+
+class MealViewController: UIViewController {
+
+    @IBOutlet weak var nameTextField: UITextField!
+    @IBOutlet weak var photoImageView: UIImageView!
+    @IBOutlet weak var raitingControl: RatingControl!
+    @IBOutlet weak var saveButton: UIBarButtonItem!
+
+    var meal: Meal?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        nameTextField.delegate = self
+
+        if let meal = meal {
+            navigationItem.title = meal.name
+            nameTextField.text   = meal.name
+            photoImageView.image = meal.photo
+            raitingControl.rating = meal.rating
+        }
+    }
+
+    @IBAction func cancel(_ sender: Any) {
+
+        let isPresentingInAddMealMode = presentingViewController is UINavigationController
```

Comment:
I would write it like this:

```Swift
     if presentingViewController != nil {
            dismiss(animated: true, completion: nil)
        } else if let navigationController = navigationController {
            navigationController.popViewController(animated: true)
        } else {
            fatalError("The MealViewController is not inside a navigation controller.")
        }
```

It shouldn't matter if presentingViewController is UINavigationController. As long as the presentingViewController is not nil, you can call dismiss.

Reply:
I understand, I didn't know that